### PR TITLE
Update parsing logic to account for recent changes to input JSON file format.

### PR DESCRIPTION
**Pull request by JCantu248**
#  #

## 🗣 Description ##

The RPT has updated the format of their JSON data that provides input to TPT-Reports. The source for TPT-Reports has been updated to reflect these changes. The `payloads` list has been nested under the `phishing_assessment` property. In addition, the assessment ID number is provided in the new format, so it is no longer necessary as a command line argument and has been removed. 

All test cases have been updated to reflect these new changes. Resolves issue #66 

## 💭 Motivation and context ##

This change is required so the PCA team can use TPT-Reports with the latest RPT JSON format. 

## 🧪 Testing ##

Test cases have been updated and verified with PyTest. 

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] *All* future TODOs are captured in issues, which are referenced
      in code comments.
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.

## ✅ Pre-merge checklist ##

- [x] Revert dependencies to default branches.
- [x] Finalize version.

## ✅ Post-merge checklist ##

- [x] Create a release.

### CODE REVIEW

**Review by nickviola — APPROVED** [review]
LGTM.

**Comment by schmelz21** [comment]
@JCantu248, @nickviola  - Can we keep the versioning process updated. I'd consider this one a version change due to the re-formatting of our input files from our customer.

File: src/tpt_reports/tpt_reports.py [hunk 1]
```diff
@@ -67,12 +66,13 @@ def parse_json(data):
     payloads_meta = {}
     try:
         if data:
-            for payload in data["payloads"]:
+            assessment_id = data["id"]
```
Comment (dav3r) [inline src/tpt_reports/tpt_reports.py]: Is it assumed that this updated code will never be run on the older JSON files?  Generally, in a situation like this, I'd use something like `assessment_id = data.get("id", "N/A")` to handle the possibility of a missing `id` field (e.g. if this were run with the older-style JSON).

File: src/tpt_reports/tpt_reports.py [hunk 2]
```diff
@@ -67,12 +66,13 @@ def parse_json(data):
...
+            for payload in data["phishing_assessment"]["payloads"]:
                 num_payloads += 1
                 payload_data = {}
                 payload_data["Payload"] = payload["payload_description"]
                 # c2_protocol requires additional space to match input file format
-                payload_data["C2 Protocol"] = payload["c2_protocol "]
+                payload_data["C2 Protocol"] = payload["c2_protocol"]
```
Comment (dav3r) [inline src/tpt_reports/tpt_reports.py:74]: Does this field name change mean that the comment above this line can now be deleted?

**Review by dav3r — COMMENTED** [review]
I have a couple of quick questions before I approve.

**Review by schmelz21 — APPROVED** [review]

File: src/tpt_reports/tpt_reports.py (hunk 2 above)
Comment (JCantu248) [reply src/tpt_reports/tpt_reports.py:74]: Yes. It has been removed in this commit 8d35110a9096931c1fc135340c73351d0755a170.

File: src/tpt_reports/tpt_reports.py (hunk 1 above)
Comment (JCantu248) [reply src/tpt_reports/tpt_reports.py]: Operators do not expect that the format would revert, but the suggestion was applied in f04114a31e9a9b8f3a2176880248a7ee2e15fa5d. 

If for whatever reason the ID is not provided, we would at least have a fallback value to apply.

File: src/tpt_reports/tpt_reports.py (hunk 2 above)
Comment (dav3r) [reply src/tpt_reports/tpt_reports.py:74]: Great, thanks!

File: src/tpt_reports/tpt_reports.py (hunk 1 above)
Comment (dav3r) [reply src/tpt_reports/tpt_reports.py]: Looks good! 👍

**Review by dav3r — APPROVED** [review]
👍 👍

**Pull request merged**